### PR TITLE
RSDK-3561 GetLatestMapInfo PR2

### PR DESCRIPTION
**Pull request by PStrutz**
JIRA Ticket: https://viam.atlassian.net/browse/RSDK-3561

viam-cartographer implementation of GetLatestMapInfo;
added mapTimestamp field to cartographerService;
keep track of changing maps when GetPointCloudMap is called

### CODE REVIEW

File: viam-cartographer.go [hunk 1]
```diff
@@ -226,12 +231,16 @@ func (cartoSvc *cartographerService) GetPosition(ctx context.Context) (spatialma
 	return vcUtils.CheckQuaternionFromClientAlgo(pose, componentReference, returnedExt)
 }
 
-// GetPointCloudMap creates a request, calls the slam algorithms GetPointCloudMap endpoint and returns a callback
+// GetPointCloudMap creates a request, **recording the time**, calls the slam algorithms GetPointCloudMap endpoint and returns a callback
```
Comment (jeremyrhyde) [inline viam-cartographer.go:234]: [nit] remove the **

**Review by jeremyrhyde — APPROVED** [review]
LGTM. One last nit comment

File: go.mod [hunk 2]
```diff
@@ -312,3 +312,7 @@ require (
 	mvdan.cc/unparam v0.0.0-20221223090309-7455f1af531d // indirect
 	nhooyr.io/websocket v1.8.7 // indirect
 )
+
+replace go.viam.com/api => /Users/patriciastrutz/Development/api
```
Comment (jeremyrhyde) [inline go.mod:316]: [note] remove before pushing

**Review by jeremyrhyde — APPROVED** [review]
LGTM. Remember to remove replace lines in go.mod before merge.

File: internal/dim-2d/dim-2d.go [hunk 3]
```diff
@@ -70,7 +70,7 @@ func ValidateGetAndSaveData(
 	var err error
 	var path string
 	paths := make([]string, 0, 1)
-	startTime := time.Now().UTC()
+	startTime := time.Now()
```
Comment (nicksanford) [inline internal/dim-2d/dim-2d.go:73]: Why are we removing this?

**Review by jeremyrhyde — APPROVED** [review]
LGTM!

**Pull request merged**